releases.md

```diff
@@ -3,6 +3,9 @@ title: Project Calico Documentation Archives
 description: Home
 layout: docwithnav
 ---
+This page contains links to all Project Calico documentation for previous
+releases, as well as to the latest release and the nightly build of
```

Comment:
Might want to say "versions" rather than "releases"?  I'm not totally sure what the best way to describe these are, but each major version (which is what the docs index on) has multiple minor version releases (which are listed in the release notes for that version of the docs).

Reply:
FWIW, I often use the term "release series".